lib/bounce_email.rb

```diff
@@ -237,5 +239,38 @@ def extract_original_message_after_delimiter(mail, delimiter_index)
       message = mail.body.to_s.split(delimiter).last
       message.split(INLINE_MESSAGE_END_DELIMITER).first.strip if message.match(INLINE_MESSAGE_END_DELIMITER)
     end
+
+    def extract_and_assign_fields_from_original_mail(mail)
+      extract_and_assign_message_id(mail) if mail.message_id.nil?
+      extract_and_assign_from(mail) if mail.from.nil?
+      extract_and_assign_to(mail) if mail.to.nil?
+      extract_and_assign_subject(mail) if mail.subject.nil?
+      mail
+    end
+
+    def extract_and_assign_message_id(mail)
+      value = extract_field(mail, 'Message-ID:')
+      mail.add_message_id(value)
+    end
+
+    def extract_and_assign_to(mail)
+      value = extract_field(mail, 'To:')
+      mail.to = value
+    end
+
+    def extract_and_assign_from(mail)
+      value = extract_field(mail, 'From:')
+      mail.from = value
+    end
+
+    def extract_and_assign_subject(mail)
+      value = extract_field(mail, 'Subject:')
+      mail.subject = value
+    end
+
+    def extract_field(mail, field_name)
+      field = mail.body.to_s.split(/[\n|\r\n]+/).detect { |line| line.match field_name }
```

Comment:
Here, in the regex, you're using a character class (square brackets). Are you sure that is intended? Or did you actually want to use a group with regular brackets - `(` and `)`? The current regex should be equivalent to "any sequence of one or more of the `\n`, `\r` or `|` symbols. Maybe this is what you want: `/(\r?\n)+/`?